packages/jni-swig-stub/src/main/jni/realm_api_helpers.cpp

```diff
@@ -689,3 +689,22 @@ sync_after_client_reset_handler(realm_sync_config_t* config, jobject after_handl
     };
     realm_sync_config_set_after_client_reset_handler(config, after_func, user_data, free_func);
 }
+
+// Explicit clean up method for releasing heap allocated data of a realm_value_t instance
+void
+realm_value_t_cleanup(realm_value_t* value) {
```

Comment:
To my point above, if we're passing a string that is going to outlive the C-API call why not build a stack-based string and pass it to Core? this will align also with the upcoming changes in the C-API where we plan to manipulate `string_data` instead of `char*` ....